### DISCUSSION TITLE
How can we reference a custom defined value in queries?

### DISCUSSION THREAD

**Discussion by baileygranam** [prisma/prisma#13436] [Q&A] [answered]
Hi all, I am trying to write a query that returns a list of resources that are available for a given range. The issue however is that I cannot seem to find any documentation on referencing custom values in the `where` statement. For now I am using a raw query but would rather not if possible.
**Current:**
```
    SELECT "Resource".*
    FROM "Resource"
    LEFT JOIN "Appointment" ON "Appointment"."resourceId" = "Resource"."id"
    WHERE "Appointment"."id" IS NULL
    OR NOT (
        "Appointment"."startAt" < ${range.endDate}
        AND ("Appointment"."startAt" + INTERVAL '1 Minute' * "Appointment"."duration") > ${range.startDate}
    );
```
**Desired:**
```
prisma.resource.findMany({
    where: {
      appointments: {
        none: {
          startAt: { lt: range.endDate },
          // How do I query:
          // ("Appointment"."startAt" + INTERVAL '1 Minute' * "Appointment"."duration") > ${range.startDate}
        },
      },
    },
  });
```

**Comment by nurul3101** [comment, marked as answer]
Hey @baileygranam 👋 
Can't you just calculate the value of startAt at the application level instead of adding an interval at the database level?

Also we have a feature request for `whereRaw` which would ideally solve this: [#5560](https://github.com/prisma/prisma/issues/5560)

**Reply by kelvindecosta** [reply]
Hey @nurul3101 

Thank you for sharing the feature request!
Could you please give an ETA for the feature?
I would really appreciate it!

**Reply by nurul3101** [reply]
We don't have a fixed ETA for this, I would suggest subscribing to the issue to be notified about it.
Also here's our [roadmap](https://www.notion.so/prismaio/Prisma-Roadmap-50766227b779464ab98899accb98295f) to have a look at what we are currently working on.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
